File: C++/Medium/WrongAnswer/statistics-from-a-large-sample_6.cpp
class Solution {
public:
    vector<double> sampleStats(vector<int>& count) {
        
        vector<double> result;
        int min = INT_MAX, max = INT_MIN, mode = 0;
        long long sum = 0;
        double mean = 0.0;
        long long total = 0;
        for (int i = 0; i < count.size(); i++) {
            if (count[i] > 0) {
                if (min > i) {
                    min = i;
                }
                if (max < i) {
                    max = i;
                }
                if (count[i] > count[mode]) {
                    mode = i;
                }
                sum += count[i] * i;
                total += count[i];
            }
        }
        mean = (double)sum / total;
        int mid = total / 2;
        int prev = 0;
        int median = 0;
        for (int i = 0; i < count.size(); i++) {
            if (count[i] > 0) {
                if (mid >= prev && mid < prev + count[i]) {
                    median = i;
                    break;
                }
                prev += count[i];
            }
        }
        if (total % 2 == 0) {
            int next = 0;
            for (int i = 0; i < count.size(); i++) {
                if (count[i] > 0) {
                    if (mid + 1 >= next && mid + 1 < next + count[i]) {
                        median = (median + i) / 2.0;
                        break;
                    }
                    next += count[i];
                }
            }
        }
        result.push_back(min);
        result.push_back(max);
        result.push_back(mean);
        result.push_back(median);
        result.push_back(mode);
        return result;
    }
};

File: Java/Medium/WrongAnswer/statistics-from-a-large-sample_6.java
class Solution {
    public double[] sampleStats(int[] count) {
        
        double[] result = new double[5];
        int N = 0;
        int max = 0;
        for (int i = 0; i < count.length; i++) {
            if (count[i] > 0) {
                if (result[0] == 0) {
                    result[0] = i;
                }
                result[1] = i;
                result[2] += i * count[i];
                N += count[i];
                if (count[i] > max) {
                    result[4] = i;
                    max = count[i];
                }
            }
        }
        result[2] /= N;
        int mid = N / 2;
        int sum = 0;
        for (int i = 0; i < count.length; i++) {
            sum += count[i];
            if (sum > mid) {
                result[3] = i;
                if (N % 2 == 0 && sum == mid + 1) {
                    for (int j = i + 1; j < count.length; j++) {
                        if (count[j] > 0) {
                            result[3] += j;
                            result[3] /= 2;
                            break;
                        }
                    }
                }
                break;
            }
        }
        return result;
    }
}